pkg/controller/machine/machine_controller.go

```diff
@@ -274,15 +278,33 @@ func (r *ReconcileMachine) isDeleteAllowed(machine *clusterv1.Machine) bool {
 	return node.UID != machine.Status.NodeRef.UID
 }
 
-func (r *ReconcileMachine) deleteNode(ctx context.Context, name string) error {
-	var node corev1.Node
-	if err := r.Client.Get(ctx, client.ObjectKey{Name: name}, &node); err != nil {
-		if apierrors.IsNotFound(err) {
-			klog.V(2).Infof("Node %q not found", name)
-			return nil
+func (r *ReconcileMachine) deleteNode(ctx context.Context, cluster *clusterv1.Cluster, name string) error {
+	if cluster == nil {
+		var node corev1.Node
+		if err := r.Client.Get(ctx, client.ObjectKey{Name: name}, &node); err != nil {
+			if apierrors.IsNotFound(err) {
+				klog.V(2).Infof("Node %q not found", name)
+				return nil
+			}
+			klog.Errorf("Failed to get node %q: %v", name, err)
+			return err
```

Comment:
This seems like it could be `return nil` / won't retry error. But I'm not sure, what was the logic behind this being an actual error?

Reply:
Removed this part, IsNotFound is checked from the caller